frontend/src/components/Sidebar.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC } from "react";
+import { HomeIcon, FolderIcon, CogIcon, ShareIcon } from "lucide-react";
+import Image from "next/image";
+import { Button } from "./ui/button";
+
+const Sidebar: FC = () => {
+  return (
+    
+      <div className="relative flex flex-col  bg-white border-r group hover:w-52 w-[56px] w-min-[56px] transition-all  duration-300 ease-in-out overflow-hidden" style={{
+        height: "calc(100vh - 64px)"
+      }}>
+        
+        <div className="flex flex-col flex-1 p-2 space-y-4">
+          <div className="grow">
+            <button className="flex items-center text-left text-gray-600 hover:bg-gray-200 p-2 rounded relative group-hover:w-full hover:text-black">
+              <HomeIcon className="h-5 w-5 flex-shrink-0" />
+              <span  className="text-left ml-8 opacity-0 group-hover:opacity-100 transition-all duration-300 ease-in-out group-hover:ml-2 whitespace-nowrap" style={{ fontSize: '14px' ,fontWeight:'500'}}>
```

Comment:
It can be done with Tailwind
```suggestion
              <span  className="text-left ml-8 opacity-0 group-hover:opacity-100 transition-all duration-300 ease-in-out group-hover:ml-2 whitespace-nowrap text-sm font-medium">
```